libsql/src/hrana/connection.rs

```diff
@@ -81,7 +81,7 @@ where
     }
 
     pub async fn prepare(&self, sql: &str) -> crate::Result<Statement<T>> {
-        let stream = self.current_stream().clone();
```

Comment:
Did the expired stream get properly cleaned up?
Is it possible to check the current stream to see if it's a live and reuse it, if it's dead open a new stream - does that improve anything?